packages/generator/tests/acceptance/test_client_generator.py

```diff
@@ -71,6 +73,45 @@ def test___command_line_args___create_client_for_all_registered_measurements___r
     )
 
 
+@patch("pathlib.Path.cwd")
+def test___command_line_args_with_registered_measurements___create_client_using_interactive_mode___renders_without_error(
+    mock_directory_out: Mock,
+    test_assets_directory: pathlib.Path,
+    tmp_path_factory: pytest.TempPathFactory,
+    measurement_service: MeasurementService,
+) -> None:
+    temp_directory = tmp_path_factory.mktemp("measurement_plugin_client_files")
+    mock_directory_out.return_value = temp_directory
+    golden_path = test_assets_directory / "example_renders" / "measurement_plugin_client"
+    filename = "non_streaming_data_measurement_client.py"
+
+    with patch(
+        "click.prompt",
+        side_effect=[
+            "1",
+            "non_streaming_data_measurement_client",
+            "NonStreamingDataMeasurementClient",
+            "x",
+        ],
+    ):
+        with pytest.raises(SystemExit) as exc_info:
```

Comment:
rather then catching the systemExit, let's use Click's testing utils -> https://dev.azure.com/ni/DevCentral/_git/Tools?path=/eng/pipeline/azdo_cli/tests/test_agents.py&version=GC3143bc8176382ff7809218dc6f83f2ad5e69a589&line=251&lineEnd=253&lineStartColumn=1&lineEndColumn=6&lineStyle=plain&_a=contents

(docs) https://click.palletsprojects.com/en/8.1.x/testing/

I usually like to make this a  fixture:
```python
@pytest.fixture()
def cli_runner():
    """Provide helper for calling the cli in tests."""
    runner = CliRunner(mix_stderr=False)
    yield functools.partial(runner.invoke, my_module.__main__.cli, catch_exceptions=False, standalone_mode=False)
```

Reply:
Updated the tests with cli_runner.